python/modem-firmware-1.3+/claim_and_provision_device.py

```diff
@@ -0,0 +1,526 @@
+    nrf_cloud_diap.print_api_result("Prov cmd CSR response", api_res, args.verbose)
+    if api_res.status_code != 201:
+        error_exit('CreateDeviceProvisioningCommand API call failed')
+
+    # get the provisioning cmd ID from the response
+    prov_id = None
+    res_json = json.loads(api_res.text)
+    if not res_json:
+        error_exit('Unexpected CreateDeviceProvisioningCommand API response')
+
+    prov_id = res_json.get('id')
+    if not prov_id:
+        error_exit('Failed to obtain provisioning cmd ID')
+
+    print(hivis_style('\nProvisioning command (CSR) ID: ' + prov_id + '\n'))
+
+    # reset the device since we disabled the modem
+    print(send_style('Resetting device'))
+    rtt_interface.reset_device(args.jlink_sn)
+    # wait for device to boot and process the command
+    print(local_style('Waiting for device to process command...'))
+    cmd_response = wait_for_cmd_status(args.api_key, dev_uuid, prov_id)
+
+    # get the CSR from the response
+    csr_txt = cmd_response.get('certificateSigningRequest').get('csr')
+    if csr_txt:
+        print(hivis_style('CSR:\n' + csr_txt + '\n'))
+
+    # process the CSR
+    modem_credentials_parser.parse_keygen_output(csr_txt)
+
+    # get the public key from the CSR
+    csr_bytes = modem_credentials_parser.csr_pem_bytes
+    try:
+        csr = OpenSSL.crypto.load_certificate_request(OpenSSL.crypto.FILETYPE_PEM,
+                                                      csr_bytes)
+        pub_key = csr.get_pubkey()
+    except OpenSSL.crypto.Error:
+        cleanup()
+        raise RuntimeError("Error loading CSR")
+
+    # create a device cert
+    print(local_style('Creating device certificate...'))
+    device_cert = create_device_cert(args.dv, csr, pub_key, ca_cert, ca_key)
+    dev_cert_pem_bytes = OpenSSL.crypto.dump_certificate(OpenSSL.crypto.FILETYPE_PEM, device_cert)
+    dev_cert_pem_str = dev_cert_pem_bytes.decode()
+    print(local_style('Dev cert: \n{}'.format(dev_cert_pem_str)))
+
+    # create provisioning command to install device cert
+    print(local_style('\nCreating provisioning command (client cert)...'))
+    api_res = nrf_cloud_diap.create_provisioning_cmd_client_cert(args.api_key, dev_uuid,
+                                                                 dev_cert_pem_str,
+                                                                 sec_tag=args.sectag)
+    nrf_cloud_diap.print_api_result("Prov cmd client cert response", api_res, args.verbose)
+    if api_res.status_code != 201:
+        error_exit('CreateDeviceProvisioningCommand API call failed')
+
+    # get the provisioning cmd ID from the response
+    res_json = json.loads(api_res.text)
+    if not res_json:
+        cleanup()
+        error_exit('Unexpected CreateDeviceProvisioningCommand API response')
+
+    prov_id = res_json.get('id')
+    if not prov_id:
+        error_exit('Failed to obtain provisioning cmd ID')
+
+    # TODO: create provisioning command to install AWS root CA?
```

Comment:
After the "Client Certificate Command," the next step should be to include the "Finish command" in the script in order to change the device status to "provisioned" in nrfcloud and restart the device after provisioning. Finish command should be added as the last command to change the status from Awaiting to Provisioned.
![image](https://github.com/nRFCloud/utils/assets/125363902/2c134e7f-fbdd-4a8a-a2cc-26c5e4224f8c)

Reply:
<img width="677" alt="image" src="https://github.com/nRFCloud/utils/assets/4138391/04bb78f7-dba3-40c4-b6e2-e87f65e64c9a">
Updated